### DISCUSSION TITLE
Error during hls_model compilation

### DISCUSSION THREAD

**Discussion by sandeep1404** [fastmachinelearning/hls4ml#739] [Q&A]
Hi, 
I am using hls4ml version 0.6.0 when I used to compile the hls_model using hls_model.compile() it throws me the below error

Writing HLS project
Done

firmware/myproject.cpp: In function ‘void myproject(input_t*, result_t*, short unsigned int&, short unsigned int&)’:
firmware/myproject.cpp:101:17: error: ‘layer15_t’ was not declared in this scope; did you mean ‘layer17_t’?
  101 |     nnet::dense<layer15_t, layer17_t, config17>(layer15_out, layer17_out, w17, b17); // qdense_1
      |                 ^~~~~~~~~
      |                 layer17_t
firmware/myproject.cpp:101:49: error: ‘layer15_out’ was not declared in this scope; did you mean ‘layer17_out’?
  101 |     nnet::dense<layer15_t, layer17_t, config17>(layer15_out, layer17_out, w17, b17); // qdense_1
      |                                                 ^~~~~~~~~~~
      |                                                 layer17_out
g++: error: myproject.o: No such file or directory
---------------------------------------------------------------------------

``
OSError                                   Traceback (most recent call last)
Cell In[31], line 1
----> 1 hls_model1.compile()

File ~/.local/lib/python3.8/site-packages/hls4ml/model/hls_model.py:546, in HLSModel.compile(self)
    544         dlclose_func.restype = ctypes.c_int
    545         dlclose_func(self._top_function_lib._handle)
--> 546     self._top_function_lib = ctypes.cdll.LoadLibrary(lib_name)
    547 finally:
    548     os.chdir(curr_dir)

File /usr/lib/python3.8/ctypes/__init__.py:451, in LibraryLoader.LoadLibrary(self, name)
    450 def LoadLibrary(self, name):
--> 451     return self._dlltype(name)

File /usr/lib/python3.8/ctypes/__init__.py:373, in CDLL.__init__(self, name, mode, handle, use_errno, use_last_error, winmode)
    370 self._FuncPtr = _FuncPtr
    372 if handle is None:
--> 373     self._handle = _dlopen(self._name, mode)
    374 else:
    375     self._handle = handle

OSError: firmware/myproject-0FfB97A6.so: cannot open shared object file: No such file or directory
``
I have no idea how to resolve this can anyone help me in resolving this issue

**Comment by vloncar** [comment]
Please submit a [new issue](https://github.com/fastmachinelearning/hls4ml/issues/new?assignees=&labels=bug&template=bug_report.md&title=) for follow-up.

**Reply by sandeep1404** [reply]
Thanks for the reply, I have submitted a new issue regarding the error
